contracts/modules/LoanOpenings/LoanOpenings.sol

```diff
@@ -13,55 +13,42 @@ import "../../mixins/InterestHandler.sol";
-        _setTarget(this.getBorrowAmountByParams.selector, target);
-
-        // TODO remove next deploy
-        _setTarget(bytes4(keccak256("getEstimatedMarginExposure(address,address,uint256,uint256,uint256,uint256,bytes")), address(0));
     }
 
     // Note: Only callable by loan pools (iTokens)
     function borrowOrTradeFromPool(
-        bytes32 loanParamsId,
+        address collateralToken,
         bytes32 loanId, // if 0, start a new loan
         bool isTorqueLoan,
         uint256 initialMargin,
         address[4] calldata sentAddresses,
-            // lender: must match loan if loanId provided
-            // borrower: must match loan if loanId provided
-            // receiver: receiver of funds (address(0) assumes borrower address)
-            // manager: delegated manager of loan unless address(0)
+        // lender: must match loan if loanId provided
+        // borrower: must match loan if loanId provided
+        // receiver: receiver of funds (address(0) assumes borrower address)
+        // manager: delegated manager of loan unless address(0)
         uint256[5] calldata sentValues,
-            // newRate: new loan interest rate
-            // newPrincipal: new loan size (borrowAmount + any borrowed interest)
-            // torqueInterest: new amount of interest to escrow for Torque loan (determines initial loan length)
-            // loanTokenReceived: total loanToken deposit (amount not sent to borrower in the case of Torque loans)
-            // collateralTokenReceived: total collateralToken deposit
-        bytes calldata loanDataBytes)
-        external
-        payable
-        nonReentrant
-        pausable
-        returns (LoanOpenData memory)
-    {
+        // newRate: new loan interest rate
+        // newPrincipal: new loan size (borrowAmount + any borrowed interest)
+        // torqueInterest: new amount of interest to escrow for Torque loan (determines initial loan length)
+        // loanTokenReceived: total loanToken deposit (amount not sent to borrower in the case of Torque loans)
+        // collateralTokenReceived: total collateralToken deposit
+        bytes calldata loanDataBytes
+    ) external payable nonReentrant pausable returns (LoanOpenData memory) {
         require(msg.value == 0 || loanDataBytes.length != 0, "loanDataBytes required with ether");
 
         // only callable by loan pools
-        require(loanPoolToUnderlying[msg.sender] != address(0), "not authorized");
+        address loanToken = loanPoolToUnderlying[msg.sender];
+        require(loanToken != address(0), "not authorized");
+        require(supportedTokens[loanToken] && supportedTokens[collateralToken], "unsupported token");
 
-        LoanParams memory loanParamsLocal = loanParams[loanParamsId];
+        LoanParams memory loanParamsLocal = getDefaultLoanParamsAndStore(loanToken, collateralToken, isTorqueLoan);
         require(loanParamsLocal.id != 0, "loanParams not exists");
```

Comment:
Remove for gas improvements as it is not a needed check. It will always pass.

Reply:
which check exactly I don't get it? the require(loanParamsLocal.id != 0, "loanParams not exists");
 ?

Reply:
ya

Reply:
It is not needed because getDefaultLoanParamsAndStore will always return a LoanParams that has an id != 0

Reply:
in current implementation of `getDefaultLoanParams` it never returns id=0; thinking for the future

Reply:
I think it is better to remove it now then if we need to throw errors if a param doesnt exist, throw it in the function that retrieves the params